server/manager/src/test/java/org/apache/accumulo/manager/upgrade/Upgrader11to12Test.java

```diff
@@ -62,198 +67,334 @@ public class Upgrader11to12Test {
   private static final Logger LOG = LoggerFactory.getLogger(Upgrader11to12Test.class);
 
   @Test
-  void upgradeDataFileCFTest() throws Exception {
+  void upgradeDataFileCF2Test() {
     Upgrader11to12 upgrader = new Upgrader11to12();
 
-    BatchWriter bw = createMock(BatchWriter.class);
-    Capture<Mutation> capturedAdd = newCapture();
-    bw.addMutation(capture(capturedAdd));
-    expectLastCall();
-
-    Capture<Mutation> capturedDelete = newCapture();
-    bw.addMutation(capture(capturedDelete));
-    expectLastCall();
-
-    replay(bw);
-
     String fileName = "hdfs://localhost:8020/accumulo/tables/12/default_tablet/A000000v.rf";
     Key k = Key.builder().row(new Text("12;")).family(DataFileColumnFamily.NAME)
         .qualifier(new Text(fileName)).build();
     Value v = new Value("1234,5678");
 
-    upgrader.upgradeDataFileCF(k, v, bw, "aTable");
+    Mutation upgrade = new Mutation(k.getRow());
+    upgrader.upgradeDataFileCF(k, v, upgrade);
 
-    StoredTabletFile stf = StoredTabletFile.of(new Path(fileName));
-    Mutation add = new Mutation(k.getRow()).at().family(DataFileColumnFamily.NAME)
-        .qualifier(stf.getMetadataText()).put(v);
-    LOG.debug("add mutation to be expected: {}", add.prettyPrint());
+    var pending = upgrade.getUpdates();
```

Comment:
could check the the size of pending is 2

Reply:
Added  with e09da28032